core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -34,6 +34,10 @@ trait ApiRequestHandler {
   def handle(request: RequestChannel.Request): Unit
 }
 
+trait BaseApis extends ApiRequestHandler {
```

Comment:
Added this trait to avoid polluting TestRaftRequestHandler with the ApisUtils dependencies

Reply:
BTW, my thought was 

```scala
trait ApisUtils extends Logging {
  this: KafkaApis =>
```

Reply:
Self types like this are an anti pattern, in my opinion. They were used years ago as a DI replacement (the cake pattern), but people have moved on since they tend to be hard to understand.